### ISSUE TITLE
Allow running a SignalK Server with a simple include

### ISSUE THREAD

**Issue by codefoster** [SignalK/signalk-server-node#548]
The fastest way I currently know of to create a new SignalK Server is to clone the `signalk-server-node` repository and then customize it. That's not too bad an option, but in my opinion I should be able to create my own Node.js project, run `npm install signalk-server`, write this code...

``` js
const SignalKServer = require('signalk-server');
let server = new SignalKServer({ settings: 'settings.json' });
server.start();
```

...and have a running server.

I realize this doesn't seem like a pressing issue because there's a workaround - just clone the repo - but I'm strongly of the opinion that this isn't the best way to start a project. Allowing it to be included into another Node project gives the developer more options such as...

- running the server conditionally
- using a process manager
- coordinating the running SignalK server with other processes on the same device

It doesn't seem the project is too far from it. I gave a shot at it, but I'm still trying to figure out some of the bootstrapping that happens with a server - loading settings files, providers, interfaces (?), modules, etc. and decided to pause the effort for now.

The first roadblock I ran into is that the Server is looking for its settings file with a path that's relative to the package root (see `lib/config/config.js:253`). I can set the settings file absolutely using the SIGNALK_NODE_SETTINGS environment variable, but that's not ideal. The absolute path is harder to maintain, and IMO, the environment variable, should not be necessary. I should be able to set it by passing in the option.

The next roadblock is this...

``` bash
{ Error: ENOENT: no such file or directory, scandir 'C:\code\azureafloat\simple-server\node_modules\signalk-server\lib/../node_modules/'
    at Object.fs.readdirSync (fs.js:893:3)
    at findModulesInDir (C:\code\azureafloat\simple-server\node_modules\signalk-server\lib\modules.js:54:6)
    at modulesWithKeyword (C:\code\azureafloat\simple-server\node_modules\signalk-server\lib\modules.js:40:5)
    at startPlugins (C:\code\azureafloat\simple-server\node_modules\signalk-server\lib\interfaces\plugins.js:140:5)
    at Object.start (C:\code\azureafloat\simple-server\node_modules\signalk-server\lib\interfaces\plugins.js:37:7)
    at C:\code\azureafloat\simple-server\node_modules\signalk-server\lib\index.js:239:60
    at C:\code\azureafloat\simple-server\node_modules\lodash\lodash.js:4925:15
    at Function.forIn (C:\code\azureafloat\simple-server\node_modules\lodash\lodash.js:12949:11)
    at startInterfaces (C:\code\azureafloat\simple-server\node_modules\signalk-server\lib\index.js:225:5)
    at C:\code\azureafloat\simple-server\node_modules\signalk-server\lib\index.js:138:7
```

I didn't try to hard to figure that one out before submitting this issue.

/cc: @taggie05

**Comment by sbender9** [comment]
Yeah, I thought this might be pretty simple, but now I'm thinking it would be a lot of work to make this happen. I don't think any of us would have time or motivation to work on it, so pull requests welcome...

Just curious, why are trying to this?

**Comment by tkurki** [comment]
Running with separate configs and embedding the server are two separate use cases. Both valid, but different. Which one are you after?

You don’t need separate clones: you can run separate instances by setting the environment variable SIGNALK_NODE_CONDFIG_DIR. You can even use a single global install from npm. Have you tried launching programmatically with this?

**Comment by codefoster** [comment]
@tkurki, at this point I'm only looking for running with separate configs, but I was thinking that my solution would cover both of the cases you mentioned - allowing for separate configs and embedding.
I didn't know about setting the SIGNALK_NODE_CONFIG_DIR variable. So would I be able to use a project that only pulls SignalK in as an npm install and not clone it? I'll try that.

**Comment by webmasterkai** [comment]
Once #595 and #555 land a user can do something like the following:

```javascript
const appRoot = require('app-root-path')
const SignalKServer = require('signalk-server')

const config = {
  appPath: appRoot.path,
  configPath: '/data/signalk-config'
}
const server = new SignalKServer({ config });
server.start();
```

**Comment by tkurki** [comment]
Fixed in #695  afaik, reopen if not.
